v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/connection/CassandraConnectionHelper.java

```diff
@@ -0,0 +1,192 @@
+   *
+   * @param connectionRequestKey The unique key identifying the connection in the pool.
+   * @return The {@link CqlSession} instance associated with the given key.
+   * @throws ConnectionException If the connection pool is not initialized or no connection is found
+   *     for the key.
+   */
+  @Override
+  public CqlSession getConnection(String connectionRequestKey) throws ConnectionException {
+    if (connectionPoolMap == null || connectionPoolMap.isEmpty()) {
+      LOG.warn("Connection pool not initialized.");
+      throw new ConnectionException("Connection pool is not initialized.");
+    }
+
+    CqlSession session = connectionPoolMap.get(connectionRequestKey);
+    if (session == null) {
+      LOG.warn("No connection found for key: {}", connectionRequestKey);
+      throw new ConnectionException(
+          "No connection available for the given key: " + connectionRequestKey);
+    }
+
+    return session;
+  }
+
+  /**
+   * Checks if the connection pool is initialized and contains connections.
+   *
+   * @return {@code true} if the connection pool is initialized and not empty; {@code false}
+   *     otherwise.
+   */
+  @Override
+  public boolean isConnectionPoolInitialized() {
+    return connectionPoolMap != null && !connectionPoolMap.isEmpty();
+  }
+
+  /**
+   * Creates a {@link CqlSession} for the given {@link CassandraShard}.
+   *
+   * @param cassandraShard The shard containing connection details.
+   * @return A {@link CqlSession} instance.
+   */
+  private CqlSession createCqlSession(CassandraShard cassandraShard) {
+    CqlSessionBuilder builder =
+        CqlSession.builder()
+            .addContactPoint(
+                new InetSocketAddress(
+                    cassandraShard.getHost(), Integer.parseInt(cassandraShard.getPort())))
+            .withAuthCredentials(cassandraShard.getUserName(), cassandraShard.getPassword())
+            .withKeyspace(cassandraShard.getKeySpaceName());
+
+    DriverConfigLoader configLoader = createConfigLoader(cassandraShard);
+    builder.withConfigLoader(configLoader);
+
+    return builder.build();
+  }
+
+  /**
+   * Generates a unique connection key for the given {@link CassandraShard}.
+   *
+   * @param shard The shard containing connection details.
+   * @return A string key uniquely identifying the connection.
+   */
+  private String generateConnectionKey(CassandraShard shard) {
+    return String.format(
+        "%s:%s/%s/%s",
+        shard.getHost(), shard.getPort(), shard.getUserName(), shard.getKeySpaceName());
+  }
+
+  /**
+   * Creates a driver configuration loader for the given {@link CassandraShard}.
+   *
+   * @param cassandraShard The shard containing configuration details.
+   * @return A {@link DriverConfigLoader} instance.
+   */
+  private DriverConfigLoader createConfigLoader(CassandraShard cassandraShard) {
+    ProgrammaticDriverConfigLoaderBuilder configLoaderBuilder =
```

Comment:
As per @bharadwaj-aditya 's suggestion to build on top of #2077 , Once that merges, you can use the `fromOptionsMap` to build the ConfigLoader.
You would not need to programmatically configure options from the file into the configLoader as it would have loaded them from the file directly.

Reply:
Address the same @VardhanThigle